### DISCUSSION TITLE
dealing with null observations and understanding qualityControl codes

### DISCUSSION THREAD

**Discussion by Kirk-A** [weather-gov/api#306] [Q&A]
I am calculating air density from my nearest weather station's latest observations, however I am frequently encountering null observation values for either temperature, barometricPressure, or relativeHumidity. Should these nulls be expected or reported?

The observation values include a **qualityControl** field, but I have not found a "decoder" for its values.  Searches within the API (https://api.weather.gov/openapi.json) for "qual" return zero results. I have have observed C, V, Z, and null qualityControl values.

What best practices do you recommend for null observations for temperature, barometricPressure, or relativeHumidity?

Thank you,
-Kirk

**Comment by scadergit** [comment]
The observation data is pass-through from [MADIS, which describes its quality control process](https://madis.ncep.noaa.gov/madis_sfc_qc_notes.shtml).
